api/tests/unit/segments/test_unit_segments_models.py

```diff
@@ -2,12 +2,88 @@
 
 import pytest
 from flag_engine.segments.constants import EQUAL, PERCENTAGE_SPLIT
+from pytest_django.fixtures import SettingsWrapper
 from pytest_mock import MockerFixture
 
 from segments.models import Condition, Segment, SegmentRule
 from segments.services import SegmentCloneService
 
 
+@pytest.mark.xfail(reason="Intentional flaky test to explain Postgres implied ordering")
+def test__ConditionManager__get_queryset__explicit_ordering_disabled__cant_guarantee_condition_ordering(
```

Comment:
I was surprised to see this test failing in CI while passing in my local, then it also failed in my local. I've added an xpass exception and it's proven to xpass/xfail at random. I find it useful for documentation, but feel free to delete it.